### PR TITLE
fix: nack timer not work as expected

### PR DESCRIPTION
**Pull request by o-u-p**

### CODE REVIEW

**Review by ibc — COMMENTED** [review]
Can you explain rationale of these changes? Was what wrong before them?

**Comment by o-u-p** [comment]
> Can you explain rationale of these changes? Was what wrong before them?

if not check 0 rtt
`if (filter == NackFilter::TIME && (nackInfo.sentAtMs == 0 || nowMs - nackInfo.sentAtMs >= this->rtt))`
will send nack too quickly.
if not stop nack timer, when nack list is empty, it will never restart, if stop the timer, it will get started here
```
if (!this->timer->IsActive())
		MayRunTimer();
```

File: worker/src/RTC/RtpStreamRecv.cpp [hunk 1]
```diff
@@ -611,12 +611,11 @@ namespace RTC
...
-		}
-
-		// Tell it to the NackGenerator.
-		if (this->params.useNack)
-		{
-			this->nackGenerator->UpdateRtt(static_cast<uint32_t>(this->rtt));
+			// Tell it to the NackGenerator.
+			if (this->params.useNack)
+			{
+				this->nackGenerator->UpdateRtt(static_cast<uint32_t>(this->rtt));
```
Comment (ibc) [inline worker/src/RTC/RtpStreamRecv.cpp:617]: Why are you just updating current RTT value in `NackGenerator` **only** if > 0.0f? This is wrong. It means that if negative then `NackGenerator` will keep using its previous given RTT value.

**Comment by ibc** [comment]
> if not stop nack timer, when nack list is empty, it will never restart, if stop the timer, it will get started here
> 
> ```
> if (!this->timer->IsActive())
> 		MayRunTimer();
> ```

Makes sense.

**Comment by ibc** [comment]
I'm including some valid changes of this PR into another one. Thanks @o-u-p

**Comment by ibc** [comment]
Closing this PR in favour of https://github.com/versatica/mediasoup/pull/1082

**Comment by ibc** [comment]
Released in mediasoup 3.11.26

**Pull request closed without merging**